### PR TITLE
Lint and Modernize

### PR DESCRIPTION
**Pull request by stephen-lazaro**
This lints and modernizes the repo, and turns the linter on for the pre-push hook. It does not do any major refactors necessary to, e.g. use const everywhere.
It also fixes a typo in the Jenkinsfile.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,14 +55,28 @@ function attachLoggersToRequest(protocol, options, callback) {
...
+          'host',
+          'protocol',
+          'auth',
+          'hostname',
+          'hash',
+          'search',
+          'query',
+          'pathname',
+          'href'
+  ));
```
Comment (jsatk) [inline index.js]: Indenting is off here.

File: index.js [hunk 2]
```diff
@@ -76,9 +88,18 @@ function attachLoggersToRequest(protocol, options, callback) {
...
+    _.assign(logInfo.response,
+          _.pick(
+            res,
+            'statusCode',
+            'headers',
+            'trailers',
+            'httpVersion',
+            'url',
+            'method'
+    ));
```
Comment (jsatk) [inline index.js]: Indenting is off here.

File: index.js [hunk 3]
```diff
@@ -57,14 +55,28 @@ function attachLoggersToRequest(protocol, options, callback) {
...
+          'href'
+  ));
 
   logInfo.request.method = req.method || 'get';
   logInfo.request.headers = req._headers;
 
-  var requestData = [];
-  var originalWrite = req.write;
-  req.write = function() {
+  let requestData = [];
```
Comment (jsatk) [inline index.js]: Can this be a `const` or no?

File: package.json [hunk 4]
```diff
@@ -12,8 +12,8 @@
   "homepage": "https://github.com/meetearnest/global-request-logger",
   "main": "index.js",
   "scripts": {
-    "lint": "eslint .",
-    "lint:fix": "eslint --fix .",
+    "lint": "./node_modules/.bin/eslint .",
+    "lint:fix": "./node_modules/.bin/eslint --fix .",
```
Comment (jsatk) [inline package.json]: Why do we need to do this?

**Comment by jsatk** [comment]
Lets chat in person about this PR. I think it's awesome and would love to see it merged, but this is an open source repo so we need to version properly, etc.

File: package.json (hunk 4 above)
Comment (stephen-lazaro) [inline package.json]: Ah, we don't, this was just me trying to debug the linter when it wasn't working.

File: index.js (hunk 3 above)
Comment (stephen-lazaro) [inline index.js]: Looks like const works.

**Comment by stephen-lazaro** [comment]
@jsatk Adjusted according to your suggestions. Let me know if anything jumps out.

File: Jenkinsfile [hunk 5]
```diff
@@ -25,7 +25,7 @@ pipeline {
       steps {
         parallel (
           "NPM:Verify": {
-            prepareNodeEnv ()
```
Comment (warhod) [inline Jenkinsfile]: How did this ever passed our tests?

Comment (warhod) [inline Jenkinsfile]: I turned on status check option for CI test... it should stop merging unless the tests passed.

Comment (stephen-lazaro) [inline Jenkinsfile]: It did not! I just wasn't paying attention and did not see that :shame:

**Pull request merged**